dags/sentinel_2/sentinel-2_gap_filler_2.py

```diff
@@ -0,0 +1,326 @@
+
+    return json.dumps(contents_dict), attributes
+
+
+def prepare_message(hook, s3_path):
+    """
+    Prepare a single message for each stac file
+    """
+    key_exists = hook.check_for_key(s3_path)
+    if not key_exists:
+        raise ValueError(f"{s3_path} does not exist")
+
+    contents, attributes = get_contents_and_attributes(hook, s3_path)
+    message = {
+        "MessageBody": json.dumps(
+            {
+                "Message": contents,
+                "MessageAttributes": attributes
+            }
+        ),
+    }
+    return message
+
+
+def publish_message(files):
+    """
+    """
+    hook = S3Hook(aws_conn_id=CONN_SENTINEL_2_SYNC)
+
+    # counter for files that no longer exist
+    failed = 0
+    sent = 0
+
+    batch = []
+
+    with ThreadPoolExecutor(max_workers=300) as executor:
+        futures = [
+            executor.submit(prepare_message, hook, s3_path)
+            for s3_path in files
+        ]
+
+        for future in as_completed(futures):
+            try:
+                batch.append(future.result())
+                if len(batch) == 10:
+                    # post_messages(batch)
+                    batch = []
+                    sent += 10
+            except Exception as exc:
+                failed += 1
+                logging.info(f"File no longer exists: {exc}")
+
+    if len(batch) > 0:
+        # post_messages(batch)
+        sent += len(batch)
+    if failed > 0:
+        raise ValueError(f"Total of {failed} files failed, Total of sent messages {sent}")
+    logging.info(f"Total of sent messages {sent}")
+
+
+def prepare_and_send_messages(limit, missing_files_chunk, **kwargs) -> None:
+    """
+    Function to retrieve the latest gap report and create messages to the filter queue process.
+
+    """
+    try:
+
+        logging.info(f"Limited: {'No limit' if limit is None else int(limit)}")
+
+        if limit is not None:
+            missing_files_chunk = missing_files_chunk[:int(limit)]
+
+        publish_message(missing_files_chunk)
+
+    except Exception as error:
+        logging.exception(error)
+        # print traceback but does not stop execution
+        traceback.print_exc()
+        raise error
+
+
+MISSING_STAC_FILES_TASK_ID = "missing_stac_files_task"
+DAG_NAME = "sentinel-2-gap-filler-TESTING"
+
+with DAG(
+        dag_id=DAG_NAME,
+        default_args=DEFAULT_ARGS,
+        schedule_interval=SCHEDULE_INTERVAL,
+        tags=["Sentinel-2", "gap-fill"],
+        max_active_runs=MAX_ACTIVE_RUNS,
+        catchup=False,
+        doc_md=__doc__,
+) as dag:
+    missing_file_chunks = get_missing_stac_files()
+
+    # TODO Do not run for lower than 50 items in the file
+    PUBLISH_MESSAGES_FOR_MISSING_SCENES = [
+        PythonOperator(
+            task_id=f"publish_messages_worker_{idx}",
+            python_callable=prepare_and_send_messages,
+            op_args=["{{ dag_run.conf.limit }}", missing_file_chunks[idx]],
```

Comment:
The TODO above explains the issue here if you run for a list smaller than the limit of workers